includes/woop/limit-payment-gateways.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Limit payment gateways.
+ *
+ * @since 1.8.1
+ * @package WC_Calypso_Bridge
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Limit payment gateways when on a plan lower than business.
+ *
+ * @param mixed $payment_gateways An array of either strings or subclasses of WC_Payment_Gateway.
+ * @return mixed Filtered array of either strings or WC_Payment_Gateway's depending on what was passed in.
+ */
+function wc_calypso_bridge_limit_payment_gateways( $payment_gateways ) {
+	if ( ! class_exists( 'Atomic_Plan_Manager' ) ||
+		! method_exists( 'Atomic_Plan_Manager', 'current_plan_slug' ) ||
+		Atomic_Plan_Manager::current_plan_slug() === Atomic_Plan_Manager::BUSINESS_PLAN_SLUG ||
+		Atomic_Plan_Manager::current_plan_slug() === Atomic_Plan_Manager::ECOMMERCE_PLAN_SLUG
+	) {
+		return $payment_gateways;
+	}
+
+	// Core WooCommerce gateways sourced from woocommerce/includes/class-wc-payment-gateways.php::init.
+	$allowed_strings = array(
```

Comment:
This [wasn't fully scoped out](https://github.com/Automattic/wp-calypso/issues/55546) before work started on it and now I think it might even need rolling back. See my thinking from yesterday [here](https://github.com/Automattic/wp-calypso/projects/114#card-69190306).

The original need stems from this post: pdgAPm-7-p2, basically we want to limit payment methods to wc pay when using non business/ecom plans in order to offset the cost of atomic and encourage uptake of wc pay. 

To reiterate the rambling from yesterday mentioned above, I suspect this is the wrong way to implement this, and likely can just be removed so long as we disable plugins on downgrade (this happens already via wpcomsh) and don't allow payment plugins be added to premium plans. I'm a little concerned about the fact you will be able to add paid plugins to premium plans from the marketplace but for now I think we can ignore it (all payment gateways are implemented as free plugins.) Premium plans won't be able to add free plugins from the marketplace. p1631779803220000/1631779179.218800-slack-C029JEQRVRT

Reply:
> for now I think we can ignore it (all payment gateways are implemented as free plugins.)

If this assumption can hold true for the future, removing would be ok. I think it is still risky that we remove this, and 3 years down the road a paid payment plugin appears in marketplace. I'd opt for a safer route where our logic isn't tied to that assumption

Reply:
Agreed